### ISSUE TITLE
Mapbox launches distance API [0]

### ISSUE THREAD

**Issue by agautsc** [codeforgreenville/trolley-tracker#78]
https://codeforgvl.imaginethat.io/trolleytracker/i/5373

**Comment by ajhodges** [comment]
Looks pretty spiffy, but as with the other APIs, I'm wondering what the limitations will be.

On that note, I have been thinking more about this lately. We can't just pass the user's location and the trolley's location to one of these things and expect to get accurate results (particularly because our route is a loop). We would need to pass the user's location, then the stop that the user is closest to, then each consecutive stop up until the trolley's current location to get an accurate time estimate. Which makes me think that all we really need is to store the estimated travel time for each edge (segment between two stops) of a route.

If we had that data in the backend, we could build an api endpoint that retrieves the "current ETA" for a given stop, which just takes the current trolley's location and adds up all the travel times in the route between the user's closest stop and the trolley's next stop.

So to summarize: user opens the app and is shown the nearest stop and when the next trolley will arrive at the stop. This data is calculated by us (upon route creation) instead of provided by an external API. Then the client apps don't have to do a bunch of heavy lifting or make a bunch of API calls, they can get the data from our own API with a single call.

Just thinking out loud. Thoughts, @bikeoid?

**Comment by bikeoid** [comment]
I agree that this needs to be server side to avoid having to duplicate logic in all the clients.

For travel time, we might want to break travel times for street segments between intersections.  That would cover cases such as Friday LunchLink where there theoretically aren't any defined stops.

In addition to the user's position, we should take into account the destination they wish to travel in.   The best stop when northbound while from main street will probably not be the nearest, but over on River/Richardson St.   Then we can suggest or calculate a walking path to that stop.
